### DISCUSSION TITLE
Making iOS IPA with AIRSDK_Windows.

### DISCUSSION THREAD

**Discussion by kanisuki** [airsdk/Adobe-Runtime-Support#2752] [General]
Hello.
I use Windows pc and adobe animate for Windows.
I made AAB with the way using adt.cfg in the AIRSDK.
"CreateAndroidAppBundle=true"

The animate also publish iOS app with Windows pc.
How is the way?
Can I make iOS  IPA  with AIRSDK_Windows?

**Comment by marchbold** [comment]
Hi, Yes you should be able to use windows to generate an IPA for iOS. You shouldn't need to change the `adt.cfg` to do this.

Just change the publish settings in Animate to use AIR for iOS, set your settings including the Apple signing certificate and provisioning profile, then publish.

**Reply by kanisuki** [reply]
Hi,　Thank you for the advice. I am trying.

**Reply by balaedit** [reply]
@marchbold How to get provisioning profile? As I just login Apple ID and click the certificate button but error is "This resource is only for developers enrolled in a developer program or members of an organization’s team in a developer program." Please Help if possible. Sorry, I am new to publish iOS publishing.

**Reply by marchbold** [reply]
@balaedit You need to sign up to the developer program and pay the registration fee in order to create and manage applications. Sounds like you haven't been through that? https://developer.apple.com/programs/enroll/

**Reply by balaedit** [reply]
Is there any way to test the app without paying registration fee for testing purposes. Finally I have to pay but meanwhile for testing.

**Reply by marchbold** [reply]
Not as far as I'm aware.

**Reply by M4rkY4** [reply]
@balaedit AIRs advantage - to build IPAs without Xcode ( even on Windows, other technologies can't do that ) has some small drawbacks, for ex. when you build IPA via Xcode, it seems that you can create a 3-day lasting debug build to test on your own device. I've never heard about AIR of being able to do that.

**Reply by marchbold** [reply]
@M4rkY4 AIR can do IPA debug builds and deploy to local test devices, you just need to setup a development certificate and provisioning profile in the apple developer console and then use the ipa-debug target for adt. This is likely a bit harder on windows but on macOS adt can install this on your connected ios device and connect your app running on the device to your debugger in an IDE to step through the code. We find this process is much faster with AIR than with other frameworks (like Unity) too.

**Reply by ajwfrost** [reply]
@M4rkY4 I'd not heard about Xcode being able to create an IPA that works for a few days, without needing accounts etc! We can take a look at that perhaps, but if they' using some internal Apple signing process/certificate then we're unlikely to be able to replicate it...

I was going to suggest whether you can try building an app for the iPhone simulator platform, using a self-signed certificate etc, but (a) I think it would still need a provisioning profile, and (b) you need a mac to use the iPhone simulator..

**Reply by M4rkY4** [reply]
@ajwfrost You need to use (dev/apple?) account, just no subscriptions seems to be required. I wasn't able to test that myself, just googled it extensively, so please take this as a pinch of salt, I hope to test in soon. A device connected by cable might be mandatory.

> Steps for Latest Xcode and iOS with Free Xcode provisioning
> 
>     Open Xcode -> Preferences -> Accounts
> 
>     add account -> login using your Apple ID
> 
>     Choose the target -> Signing and Capabilities
> 
>     In the team choose Personal team
> 
>     Make sure Automatic signing is enabled
> 
>     Connect your device and choose it in Xcode
> 
>     Run the project it'd probably fail
> 
>     Open the iOS device settings -> General -> Device Management -> Click Trust for the app
>

**Reply by marchbold** [reply]
I didn't know this was possible either but seems you can use just an apple account to create a fairly limited local testing app through Xcode. Can't use a heap of services with it though, eg IAP, Game Center, push notifications.

Some decent info about it here:  https://stackoverflow.com/questions/30973799/ios-9-new-feature-free-provisioning-run-your-app-on-a-device-just-with-your-ap/32249026#32249026

**Reply by ajwfrost** [reply]
Thanks both. From the look of this it may still require an "iPhone developer" certificate .. which would presumably still mean a developer account is needed?

I might set up a new profile on a macbook here and see whether I can get this working with my personal apple ID. If we can check the generated IPA/provisioning profile we can then confirm what certificate is used and see whether it's possible to do something here with a self-signed certificate instead.....

cheers
